### PR TITLE
Refactor monologue and SWE agent to use the messages in state history

### PR DESCRIPTION
**Pull request by enyst**
This PR updates monologue agent to the current architecture. It's a follow-up to #1847.
Necessary for the #1771 to start working with the right components.

@rbren ~~I assume we need to take a look at any agent who currently has some sort of own history or copies, for the session management.~~ Done.
- updated monologue
- updated swe agent
- planner and micro are well-behaved already

Please note that this PR is cutting off monologue from memory condenser, since it doesn't maintain its own history of events anymore, thus can't directly replace it. This can be reworked and re-connected after 1810 and 1771.

### CODE REVIEW

File: agenthub/monologue_agent/agent.py [hunk 1]
```diff
@@ -87,7 +42,7 @@ class MonologueAgent(Agent):
     """
 
     _initialized = False
-    monologue: ShortTermHistory
+    monologue: list[dict[str, str]]  # initial thoughts ONLY
```
Comment (neubig) [inline agenthub/monologue_agent/agent.py:45]: This should probably be renamed `initial_thoughts` (in which case the comment can be removed)

**Review by neubig — COMMENTED** [review]
Mostly LGTM, but I had one naming suggestion.

File: agenthub/monologue_agent/agent.py (hunk 1 above)
Comment (enyst) [reply agenthub/monologue_agent/agent.py]: Fixed

**Review by neubig — APPROVED** [review]
Thanks!

**Comment by enyst** [comment]
Definitely. Not blocking, but it's quite likely there'll be some conflicts. I'll take care of that.

**Pull request merged**